sdk/storage/inc/datalake/datalake_options.hpp

```diff
@@ -0,0 +1,1037 @@
+       *        limited. If the number of filesystems to be returned exceeds
+       *        this limit, a continuation token is returned in the response
+       *        header x-ms-continuation. When a continuation token is returned
+       *        in the response, it must be specified in a subsequent invocation
+       *        of the list operation to continue listing the filesystems.
+      */
+      std::string Continuation;
+
+      /**
+       * @brief An optional value that specifies the maximum number of items to
+       *        return. If omitted or greater than 5,000, the response will
+       *        include up to 5,000 items.
+      */
+      uint32_t MaxResults = 0;
+    };
+
+    /**
+     * @brief Optional parameters for DataLakeFileSystemClient::Create
+     */
+    struct CreateFileSystemOptions : public SharedRequestOptions
+    {
+      /**
+       * @brief User-defined properties to be stored with the filesystem, in
+       *        the format of a comma-separated list of name and value pairs
+       *        "n1=v1, n2=v2, ...".
+       *        Note that the string may only contain ASCII characters in the
+       *        ISO-8859-1 character set.
+       * @remark You can leverage SerializeDataLakePropertiesString to create
+       *         the properties string from an std::map.
+       */
+      std::string Properties;
+    };
+
+    /**
+     * @brief Optional parameters for DataLakeFileSystemClient::Delete
+     */
+    struct DeleteFileSystemOptions : public SharedRequestOptions
+    {
+      /**
+       * @brief A date and time value. Specify this header to perform the
+       *        operation only if the resource has been modified since the
+       (        specified date and time.
+      */
+      std::string IfModifiedSince;
+
+      /**
+       * @brief A date and time value. Specify this header to perform the
+       *        operation only if the resource has not been modified since
+       *        the specified date and time.
+      */
+      std::string IfUnmodifiedSince;
+    };
+
+    /**
+     * @brief Optional parameters for DataLakeFileSystemClient::GetProperties
+     */
+    struct GetFileSystemPropertiesOptions : public SharedRequestOptions
+    {
+    };
+
+    /**
+     * @brief Optional parameters for DataLakeFileSystemClient::SetProperties
+     */
+    struct SetFileSystemPropertiesOptions : public SharedRequestOptions
+    {
+      /**
+       * @brief User-defined properties to be stored with the filesystem, in
+       *        the format of a comma-separated list of name and value pairs
+       *        "n1=v1, n2=v2, ...".
+       *        Note that the string may only contain ASCII characters in the
+       *        ISO-8859-1 character set.
+       * @remark You can leverage SerializeDataLakePropertiesString to create
+       *         the properties string from an std::map.
+       */
+      std::string Properties;
```

Comment:
This should be a `std::map<std::string, std::string>`

Reply:
Should this be called `Properties` or `Metadata`?

Reply:
Offical doc is calling this as properties and this is aligned with JS.